### PR TITLE
Update javadoc tag to new scaladoc tags.

### PR DESCRIPTION
**Pull request by pocket7878**
review @vladureche 

https://issues.scala-lang.org/browse/SI-6144

### CODE REVIEW

File: src/scaladoc/scala/tools/nsc/doc/ScaladocGlobal.scala [hunk 1]
```diff
@@ -33,6 +34,29 @@ trait ScaladocGlobalTrait extends Global {
       log(s"Suppressing error involving $root: $ex")
     }
   }
+
+  def replaceInheritDocToInheritdoc(docStr: String):String  = {
+    docStr.replaceAll("""\{@inheritDoc\p{Zs}*\}""", "@inheritdoc")
+  }
+
+  override def cookedDocComment(sym: Symbol, docStr: String = ""): String = {
```
Comment (VladUreche) [inline src/scaladoc/scala/tools/nsc/doc/ScaladocGlobal.scala]: @pocket7878, wasn't this here already? Did you move it from somewhere else?

**Comment by VladUreche** [comment]
Thanks for the patch @pocket7878!

File: src/scaladoc/scala/tools/nsc/doc/ScaladocGlobal.scala (hunk 1 above)
Comment (VladUreche) [inline src/scaladoc/scala/tools/nsc/doc/ScaladocGlobal.scala]: It looks like you copy-pasted this code from `DocComments.scala`, but forgot to remove the code there. Would it be possible to remove one copy or the other? The DRY (= Don't Repeat Yourself) principle says we should only have one copy of the method.

File: src/scaladoc/scala/tools/nsc/doc/ScaladocGlobal.scala [hunk 2]
```diff
@@ -33,6 +34,29 @@ trait ScaladocGlobalTrait extends Global {
...
+    cookedDocComments.getOrElseUpdate(sym, {
+      var ownComment = if (docStr.length == 0) docComments get sym map (_.template) getOrElse ""
+      else DocComment(docStr).template
+      ownComment = replaceInheritDocToInheritdoc(ownComment)
```
Comment (pocket7878) [inline src/scaladoc/scala/tools/nsc/doc/ScaladocGlobal.scala]: I want to replace {@inheritDoc} to @inheritdoc and delegate to scala compilers @inheritdoc handler.

Comment (VladUreche) [inline src/scaladoc/scala/tools/nsc/doc/ScaladocGlobal.scala]: Sure, but why not do this in `DocComments.scala`? Why do you need to copy-paste the code in `ScaladocGlobal.scala`?

Comment (pocket7878) [inline src/scaladoc/scala/tools/nsc/doc/ScaladocGlobal.scala]: I was thinking this conversion is only required under scaladoc.
And I think DocComments.scala is under compiler section. So I want  to enable this feature only in scaladoc. 

I'll add this feature directly to DocComments.scala.

**Comment by VladUreche** [comment]
The updated version LGTM, thanks a lot @pocket7878!
I'll merge the patch as soon as the pull request build is done.

**Pull request merged**